contracts/actions/ActionBase.sol

```diff
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity >=0.7.6;
+pragma abicoder v2;
+
+abstract contract ActionBase {
+  enum ActionType {
+    FLASHLOAN,
```

Comment:
I see what You tried to do with this, 
I think there ate two kinds of operations

1) with callbacks
2) without callbacks

but maybe there is really only one, and fact that operaation makes a callback is its internal detail as long as valle address is passed as parameter